lib/mayaUsd/utils/util.h

```diff
@@ -556,6 +556,41 @@ MString convert(const TfToken& token);
 MAYAUSD_CORE_PUBLIC
 std::string convert(const MString&);
 
+MAYAUSD_CORE_PUBLIC
+MDagPath getDagPath(const MFnDependencyNode& depNodeFn, const bool reportError = true);
+
+MAYAUSD_CORE_PUBLIC
+MDagPathMap<SdfPath> getDagPathMap(const MFnDependencyNode& depNodeFn, const SdfPath& usdPath);
+
+MAYAUSD_CORE_PUBLIC
+VtIntArray shiftIndices(const VtIntArray& array, int shift);
+
+template <typename T>
+VtValue PushFirstValue(VtArray<T> arr, const T& value)
+{
+    arr.resize(arr.size() + 1);
+    std::move_backward(arr.begin(), arr.end() - 1, arr.end());
+    arr[0] = value;
+    return VtValue(arr);
+}
+
+MAYAUSD_CORE_PUBLIC
+VtValue pushFirstValue(const VtValue& arr, const VtValue& defaultValue);
+
+template <typename T>
+VtValue PopFirstValue(VtArray<T> arr)
+{
+    std::move(arr.begin() + 1, arr.end(), arr.begin());
+    arr.pop_back();
+    return VtValue(arr);
+}
+
+MAYAUSD_CORE_PUBLIC
+VtValue popFirstValue(const VtValue& arr);
+
+MAYAUSD_CORE_PUBLIC
+bool containsUnauthoredValues(const VtIntArray& indices);
```

Comment:
I don't think any of these make sense as public utility methods. They were all previously private helper functions for working with primvar data, and they should probably stay that way. I don't see them being useful on their own.

Reply:
As you probably have seen by now, meshWriter_Primvars is eventually removed. I originally wanted to keep them in meshWriteUtils but that didn't seem like a good place, hence util.h 

They seem to fit somewhere in `lib\usd\utils` now that this directory exist.

Reply:
Where do they actually end up at the end of all of this though? If they're in `lib/mayaUsd/utils/util.cpp` just as helper functions there, then that's fine, but I don't think they should be made publicly available in any header file.

Reply:
I originally didn't also like to have these exposed in `mayaUsd/utils/util.h`. Now that I think more, this can go back inside an anonymous namespace in meshWriteUtils.